hybrid_planning_demo/src/global_mtc_planner.cpp

```diff
@@ -0,0 +1,132 @@
+#include <hybrid_planning_demo/global_mtc_planner.h>
+
+// MTC Stages
+#include <moveit/task_constructor/stages/compute_ik.h>
+#include <moveit/task_constructor/stages/connect.h>
+#include <moveit/task_constructor/stages/current_state.h>
+#include <moveit/task_constructor/stages/generate_grasp_pose.h>
+#include <moveit/task_constructor/stages/generate_pose.h>
+#include <moveit/task_constructor/stages/generate_place_pose.h>
+#include <moveit/task_constructor/stages/modify_planning_scene.h>
+#include <moveit/task_constructor/stages/move_relative.h>
+#include <moveit/task_constructor/stages/move_to.h>
+#include <moveit/task_constructor/stages/predicate_filter.h>
+#include <moveit/task_constructor/solvers/cartesian_path.h>
+#include <moveit/task_constructor/solvers/pipeline_planner.h>
+
+namespace hybrid_planning_demo
+{
+using namespace std::chrono_literals;
+using namespace moveit::task_constructor;
+GlobalMTCPlannerComponent::GlobalMTCPlannerComponent(const rclcpp::NodeOptions& options)
+  : GlobalPlannerComponent(options)
+{
+}
+
+bool GlobalMTCPlannerComponent::init()
+{
+  return true;
+}
+
+moveit_msgs::msg::MotionPlanResponse
+GlobalMTCPlannerComponent::plan(const moveit_msgs::msg::MotionPlanRequest& planning_problem)
+{
+  // Result
+  moveit_msgs::msg::MotionPlanResponse planning_solution;
+  planning_solution.error_code.val = planning_solution.error_code.SUCCESS;
+  planning_solution.group_name = "ur_manipulator";
```

Comment:
there are still several hard-coded variables as this task is really just meant as an example